### ISSUE TITLE
Add 'magnifier' actionbar icon to more screens or introduce longclick shortcut

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#11897]
### Tell us your idea!

When switching between caches, we mostly use(d) the magnifier in the header row of the "good old main menu". The main menu is gone (sad enough, esp. as behavior of the "Back" button changed completely, as c:geo has no "fall back page" anymore, but will terminate from wherever you are), and so is the quick possibility of using the magnifier's history function: Now you have to click "More" on the bottom and then "Login dashboard" on the top to get tot the magnifier ... annoying.

### What solution would you suggest?

Please add the magnifier icon (and it's history function) to a) the "More" page and b) the "Search" page, to be able to reach the history of viewed caches quick again.

**Comment by fm-sys** [comment]
I understand the current implementation is not yet perfect. But hey, it's a nightly build ;-)

But before stepping into this issue, the general role of the "more" activity needs to be clarified first. See https://github.com/cgeo/cgeo/issues/11904

**Comment by fm-sys** [comment]
But another consistent idea:
If we want to implement some long-click shortcuts into the bottom navigation (see  #11905), we could show a search view on the currently enabled activity if "search" gets long clicked. (would be quite easy to implement and even saves one more click)

**Comment by Lineflyer** [comment]
> But even if - it's not really worth spending a complete screen for, isn't it?

I deem it quite important to have a neutral "landing page" but not directly start into any activity which might be undesired by users.
So having a dashboard is a must IMHO...but we can still improve it and add/remove/restructure the information.

**Comment by moving-bits** [comment]
> > But even if - it's not really worth spending a complete screen for, isn't it?
> 
> I deem it quite important to have a neutral "landing page" but not directly start into any activity which might be undesired by users. So having a dashboard is a must IMHO...but we can still improve it and add/remove/restructure the information.

Agree with that, and additionally it gives us a "home base" to display important status notifications inside the app, e. g. available updates or connection errors.

**Comment by fm-sys** [comment]
If you don't want to start with the dashboard, you can create a home screen shortcut directly pointing to e.g. the live map...

**Comment by fm-sys** [comment]
I've renamed the issue a bit to better correspond to the current ideas

**Comment by fm-sys** [comment]
> you will never come back to it by just using the back button - which is IMHO the definition of a landing page: Hit back, until you "landed" again, on this special page which implicitly tells you "next back button will finish the app". This, IMHO, c:geo currently has no landing page at all.

Exactly. And that's the definition of having a bottom navigation. "Bottom navigation bars allow movement between primary destinations in an app." ([source](https://material.io/components/bottom-navigation)) All five destinations are top level. There's nothing which comes above.

> Meanwhile it irritates a bit, esp. as the bottom navigation is not everywhere active (yet?).

Yes, this is really something which needs to and will be improved. The bottom navigation bar is currently shown/hidden depending on the view hierarchy. Only if the activity is top-level, the view will be made visible. 

But based on material.io the bottom navigation should even appear on all screen. And view hierarchy should be removed completely. But there's still much work which needs to be done first to even just think about it...

**Comment by Lineflyer** [comment]
@MagpieFourtyTwo 
Thanks for your feedback.
Please kindly put the new topics into a new issue. While reading through I think its unrelated to the magnifier icon, isn't it?

**Comment by SammysHP** [comment]
> Instead of opening a (newly introduced) popup, can we jump to the quick search which automatically opens a list of recent caches and additionally allows you to perform a search? If yes, would it make sense to add this to the search icon?

_Originally posted by @SammysHP in https://github.com/cgeo/cgeo/issues/12687#issuecomment-1030832163_

---

>We could introduce a new transparent "quick search" activity that is started with FLAG_ACTIVITY_NO_HISTORY. Then it can be started from every activity (e.g. by long press on search in the bottom navigation).

_Originally posted by @SammysHP in https://github.com/cgeo/cgeo/issues/12687#issuecomment-1030839079_

**Comment by MagpieFourtyTwo** [comment]
Suppose we can close this one, as we have the view history via long click on the search main activity icon now?

**Comment by SammysHP** [comment]
No, this should be kept open because it is about more than just the history. Adding a "simple", global search (like the "magnifier action bar icon" on the home screen) to more or all activities allows much more (direct access to caches via GC code, keyword search in offline caches, sometimes direct trackable access etc).

**Comment by fm-sys** [comment]
It depends on the use case. Is the cache history enough, or would you like to see a possibility to enter a specific name/geocode in the search shortcut as well? OTOH the dashboard search is also quite reachable anyway... (only one click more)

**Comment by fm-sys** [comment]
> Adding a "simple", global search (like the "magnifier action bar icon" on the home screen) to more or all activities allows much more

This is exactly what the bottom navigation is not for. The bottom navigation allows fast switching between activities, and should avoid that more complex functionality needs to be added to multiple activities.

Also after reading @MagpieFourtyTwo's second statement, I would be in favor of closing this issue.

Sometimes less is more. The current long-click feature is a lesson in simplicity and works perfectly for the described usecase. Making it more complex would IMHO rather decrease usability again.

**Comment by Lineflyer** [comment]
I would also tend to close this issue. The history shortcut is there and all other searches are one click away on 'Search' in bottom nav.

What could be added in 'Search' could be this offline keyword quick search which is currently only on the dashboard. But that would be another issue.

**Comment by Lineflyer** [comment]
> What could be added in 'Search' could be this offline keyword quick search which is currently only on the dashboard. But that would be another issue.

Forget it...its already there in the keyword search.

**Comment by SammysHP** [comment]
> This is exactly what the bottom navigation is not for. The bottom navigation allows fast switching between activities, and should avoid that more complex functionality needs to be added to multiple activities.

I don't understand this. Right now we have long-click shortcuts on two out of five buttons.

**Comment by Lineflyer** [comment]
Speaking of quick access (slightly offtopic for this issue...doe we have an issue for shortcuts on the home screen?):

I showed this mock-up to some users, who asked for more flexibility in directly accessing functions. They liked it.
Not sure though if technically feasible and meaningful.

The idea is, to provide a custom option to add 3-4 individual shortcuts (to some activities like "Go to" as well as to a specific geocode or what else could be desired by users).
Initially it would only be the "+" to add a shortcut. Existing shortcuts are accessible by short click, changeable/deletable by long click.

What do others think about such use of the home screen?


![image](https://user-images.githubusercontent.com/949669/155892943-15353bc3-948f-4e7b-9be7-b0fae4c5d189.png)

**Comment by moving-bits** [comment]
We are tracking a similar idea under https://github.com/cgeo/cgeo/issues/12790#issuecomment-1049233581, and I have a first draft for it ready locally. Will post a screenshot when it's a bit more advanced. (And, WRT to the 👎: it's totally optional, if you don't select any quick launch button, home screen will stay the same as today.)

**Comment by Lineflyer** [comment]
@moving-bits 
That issue was, what I was looking for, will delete my comment here to focus in one thread.
